splinterd/src/main.rs

```diff
@@ -513,164 +399,82 @@ fn get_transport(
+                return Err(GetTransportError::CertError(
+                    "Must provide a valid client key path".into(),
+                ));
+            }
 
             let ca_file = {
                 if matches.is_present("insecure") {
-                    warn!("Starting TlsTransport in insecure mode");
                     None
                 } else {
-                    let ca_file = matches
-                        .value_of("ca_file")
-                        .map(String::from)
-                        .or_else(|| config.ca_certs())
-                        .or_else(|| {
-                            let cert_dir_path = Path::new(&cert_dir);
-                            let ca_path = cert_dir_path.join(CA_PEM);
-                            if !ca_path.is_file() {
-                                error!("CA file not found: {:?}", ca_path);
-                                return None;
-                            }
-                            let ca_file: Option<String> =
-                                cert_dir_path.join(CA_PEM).to_str().map(ToOwned::to_owned);
-                            ca_file
-                        })
-                        .ok_or_else(|| {
-                            GetTransportError::CertError(
-                                "must provide a valid file containing ca certs".into(),
-                            )
-                        })?;
-                    Some(ca_file)
-                }
-            };
-
-            let ca_file_log = {
-                if let Some(ca_file) = &ca_file {
+                    let ca_file = config.ca_certs();
+                    if !Path::new(&ca_file).is_file() {
+                        return Err(GetTransportError::CertError(
+                            "Must provide a valid file containing ca certs".into(),
+                        ));
+                    }
                     match fs::canonicalize(&ca_file)?.to_str() {
-                        Some(ca_path) => ca_path.to_string(),
+                        Some(ca_path) => Some(ca_path.to_string()),
                         None => {
                             return Err(GetTransportError::CertError(
                                 "CA path is not a valid path".to_string(),
                             ))
                         }
                     }
-                } else {
-                    "insecure".to_string()
                 }
             };
 
-            let log_value = format!(
-                "transport_type: tls, ca_certs: {:?}, client_cert: {:?}, \
-                 client_key: {:?}, server_cert: {:?}, server_key: {:?}",
-                ca_file_log,
-                fs::canonicalize(client_cert.clone())?,
-                fs::canonicalize(client_key_file.clone())?,
-                fs::canonicalize(server_cert.clone())?,
-                fs::canonicalize(server_key_file.clone())?,
+            config.log_transport_as_debug(
+                PathBuf::from(config.cert_dir()),
```

Comment:
When this is logged, it should also be an absolute path

Reply:
Addressed.